pyo3-ffi/src/cpython/code.rs

```diff
@@ -177,20 +136,24 @@ pub struct PyCodeObject {
     pub co_linetable: *mut PyObject,
     pub co_weakreflist: *mut PyObject,
     #[cfg(not(Py_3_12))]
-    pub _co_code: *mut PyObject,
+    _co_code: *mut PyObject,
     #[cfg(not(Py_3_12))]
-    pub _co_linearray: *mut c_char,
+    _co_linearray: *mut c_char,
     #[cfg(Py_3_13)]
+    #[allow(
+        private_interfaces,
+        reason = "field is public, but the type is opaque and private"
```

Comment:
we don't have the msrv for this

Reply:
Also, isn't `PyCodeObject` itself opaque? I'd rather get make it opaque than deal with all this conditional compilation.

Reply:
Yes, you're right:

https://docs.python.org/3/c-api/code.html#c.PyCodeObject

> The C structure of the objects used to describe code objects. The fields of this type are subject to change at any time.

I read that as "including in patch releases", so we should not expose this structure.